### ISSUE TITLE
Tensorflow would sometimes get rather slow during training

### ISSUE THREAD

**Issue by FduJyy** [tensorflow/tensorflow#14528]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: pip
- **TensorFlow version (use command below)**: 1.4
- **Python version**: 3.5
- **CUDA/cuDNN version**: CUDA 8.0 / cuDNN 6.0
- **GPU model and memory**: GTX1060 6GB
### Describe the problem
When I train my network, at first it works properly, however sometimes it gets rather slower (about 10 times slower than normal speed, shown below). When I type nvidia-smi to check the GPU state, I find the GPU-Util is rather low (2-3%)
The normal speed is about 1 minute per 100 steps while it would sometimes cost 10 minminutes per 100 steps. 
I used to use Tensorflow v1.1 and did not find this problem. However I updated Tensorflow to v1.4 several days ago and this problem occurred.

### Source code / logs
> 2017-11-12 23:43:55  Step: 770900, loss: 1.6261, accuracy: 68.7500%
2017-11-12 23:45:37  Step: 771000, loss: 1.4571, accuracy: 65.6250%
2017-11-12 23:48:37  Step: 771100, loss: 1.3790, accuracy: 73.4375%
2017-11-12 23:49:50  Step: 771200, loss: 1.6311, accuracy: 57.8125%
2017-11-12 23:51:11  Step: 771300, loss: 1.3248, accuracy: 67.1875%
2017-11-12 23:52:31  Step: 771400, loss: 1.6924, accuracy: 64.0625%
2017-11-12 23:53:44  Step: 771500, loss: 1.3097, accuracy: 67.1875%
2017-11-12 23:55:02  Step: 771600, loss: 1.7720, accuracy: 64.0625%
2017-11-12 23:56:15  Step: 771700, loss: 1.5915, accuracy: 68.7500%
2017-11-12 23:57:28  Step: 771800, loss: 1.7489, accuracy: 60.9375%
2017-11-12 23:58:50  Step: 771900, loss: 1.2855, accuracy: 68.7500%
2017-11-13 00:00:03  Step: 772000, loss: 1.3734, accuracy: 65.6250%
2017-11-13 00:01:16  Step: 772100, loss: 1.4637, accuracy: 70.3125%
2017-11-13 00:02:38  Step: 772200, loss: 1.6229, accuracy: 60.9375%
2017-11-13 00:03:51  Step: 772300, loss: 1.3330, accuracy: 65.6250%
2017-11-13 00:05:04  Step: 772400, loss: 1.7729, accuracy: 56.2500%
2017-11-13 00:06:37  Step: 772500, loss: 1.2806, accuracy: 65.6250%
2017-11-13 00:07:59  Step: 772600, loss: 1.0916, accuracy: 70.3125%
2017-11-13 00:09:31  Step: 772700, loss: 1.4440, accuracy: 73.4375%
2017-11-13 00:10:43  Step: 772800, loss: 0.9338, accuracy: 76.5625%
2017-11-13 00:13:03  Step: 772900, loss: 1.6038, accuracy: 57.8125%
2017-11-13 00:14:15  Step: 773000, loss: 1.2853, accuracy: 64.0625%
2017-11-13 00:16:43  Step: 773100, loss: 1.1761, accuracy: 75.0000%
2017-11-13 00:17:56  Step: 773200, loss: 1.0927, accuracy: 68.7500%
2017-11-13 00:23:23  Step: 773300, loss: 1.3884, accuracy: 73.4375%
2017-11-13 00:24:37  Step: 773400, loss: 1.3729, accuracy: 67.1875%
2017-11-13 00:25:50  Step: 773500, loss: 1.0257, accuracy: 76.5625%
2017-11-13 00:29:01  Step: 773600, loss: 1.4766, accuracy: 68.7500%
2017-11-13 00:38:35  Step: 773700, loss: 1.3401, accuracy: 68.7500%
2017-11-13 00:47:49  Step: 773800, loss: 1.4850, accuracy: 68.7500%
2017-11-13 00:57:39  Step: 773900, loss: 1.2555, accuracy: 68.7500%
2017-11-13 01:06:59  Step: 774000, loss: 1.6202, accuracy: 60.9375%
2017-11-13 01:16:28  Step: 774100, loss: 1.4673, accuracy: 59.3750%
2017-11-13 01:26:11  Step: 774200, loss: 1.3678, accuracy: 67.1875%
2017-11-13 01:35:38  Step: 774300, loss: 1.2268, accuracy: 67.1875%
2017-11-13 01:45:14  Step: 774400, loss: 1.5675, accuracy: 60.9375%
2017-11-13 01:55:00  Step: 774500, loss: 1.7380, accuracy: 59.3750%
2017-11-13 02:04:42  Step: 774600, loss: 1.9971, accuracy: 56.2500%
2017-11-13 02:15:26  Step: 774700, loss: 1.9669, accuracy: 54.6875%
2017-11-13 02:28:05  Step: 774800, loss: 1.3680, accuracy: 67.1875%
2017-11-13 02:39:03  Step: 774900, loss: 1.1381, accuracy: 76.5625%

**Comment by rohan100jain** [comment]
Its a little challenging to debug without seeing the model code you're running. Could you please share that and then might be able to help you better.

**Comment by FduJyy** [comment]
@rohan100jain The model I run is Inception_v1 provided here [https://github.com/tensorflow/models/blob/master/research/slim/nets/inception_v1.py](url)
The code is like:
	

```
        with tf.name_scope('input'):
		tra_image_batch, tra_label_batch = input_data3.getbatch(BATCH_SIZE, 'inception_v1',
			224, True, 0)
		val_image_batch, val_label_batch = input_data3.getbatch(test_batchsize, 'inception_v1',
			224, False, 0)
	x = tf.placeholder(tf.float32, shape=[None, IMG_W, IMG_H, 3])
	y_ = tf.placeholder(tf.int32, shape=[None])
	is_training = tf.placeholder(tf.bool, shape=[])
	with slim.arg_scope(inception_v1.inception_v1_arg_scope()):
		logits, end_points = inception_v1.inception_v1(x, 1001, is_training=is_training)
	loss = tools.loss(logits, y_)
	accuracy = tools.accuracy(logits, y_)
	top5_accuracy = tools.top5_accuracy(logits, y_)
        train_op = tools.optimize_m(loss, learning_rate1, learning_rate2, my_global_step)
	try:
                for step in np.arange(MAX_STEP):
			if coord.should_stop():
					break
			tra_images,tra_labels = sess.run([tra_image_batch, tra_label_batch])
			_, tra_loss, tra_acc = sess.run([train_op, loss, accuracy], feed_dict={x:tra_images, y_:tra_labels, is_training:True})
			if step % 100 == 0 or step == MAX_STEP:
				nowtime = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())
				print (nowtime+'  Step: %d, loss: %.4f, accuracy: %.4f%%' %(step+global_step, tra_loss, tra_acc))

```

**Comment by AlexGfocus** [comment]
I have same issue, training become very slow after updating to TF 1.4. But when I update tensorflow-gpu to version 1.4.0 too (pip install tensorflow-gpu==1.4.0), it works.

**Comment by drpngx** [comment]
@tfboyd @sguada Is that something we know about? Any way to debug?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assigneee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 171 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 186 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 201 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 216 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 231 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tfboyd** [comment]
To close this and likely not with a great end.  I wrote this script a long time ago and it watches nvidia-smi for any type of GPU slow down.  I doubt that is the issue, but it was my issue yesterday and I was reminded about this ticket.  All it does is run NVIDIA-SMI, grep for a very basic Hardware slowdown as well as keep a log of each full `nvidia-smi` result.  

https://github.com/tfboyd/tf-tools/blob/master/benchmark/multi_gpu/stats_monitor.sh
